examples/dmd/parametric_dmdc_heat_conduction.cpp

```diff
@@ -0,0 +1,1070 @@
+
+            dmd_prediction_timer.Start();
+
+            // 14. Predict the state at t_final using DMDc.
+            if (myid == 0)
+            {
+                std::cout << "Predicting temperature using DMDc" << std::endl;
+            }
+
+            CAROM::Vector* result_u = dmd_u->predict(ts[0]);
+            Vector initial_dmd_solution_u(result_u->getData(), result_u->dim());
+            u_gf.SetFromTrueDofs(initial_dmd_solution_u);
+
+            dmd_prediction_timer.Stop();
+
+            VisItDataCollection dmd_visit_dc(io_dir +
+                                             "/parametric_dmdc_Heat_Conduction_ROM", pmesh);
+            dmd_visit_dc.RegisterField("temperature", &u_gf);
+
+            if (visit)
+            {
+                dmd_visit_dc.SetCycle(0);
+                dmd_visit_dc.SetTime(0.0);
+                dmd_visit_dc.Save();
+            }
+
+            delete result_u;
+
+            if (visit)
+            {
+                for (int i = 1; i < ts.size(); i++)
+                {
+                    if (i == ts.size() - 1 || (i % vis_steps) == 0)
+                    {
+                        result_u = dmd_u->predict(ts[i]);
+                        Vector dmd_solution_u(result_u->getData(), result_u->dim());
+                        u_gf.SetFromTrueDofs(dmd_solution_u);
+
+                        dmd_visit_dc.SetCycle(i);
+                        dmd_visit_dc.SetTime(ts[i]);
+                        dmd_visit_dc.Save();
+
+                        delete result_u;
+                    }
+                }
+            }
+
+            result_u = dmd_u->predict(t_final);
+
+            // 15. Calculate the relative error between the DMDc final solution and the true solution.
+            Vector dmd_solution_u(result_u->getData(), result_u->dim());
+            Vector diff_u(true_solution_u.Size());
+            subtract(dmd_solution_u, true_solution_u, diff_u);
+
+            double tot_diff_norm_u = sqrt(InnerProduct(MPI_COMM_WORLD, diff_u, diff_u));
+            double tot_true_solution_u_norm = sqrt(InnerProduct(MPI_COMM_WORLD,
+                                                   true_solution_u, true_solution_u));
+
+            if (myid == 0)
+            {
+
+                std::cout << "Relative error of DMDc temperature (u) at t_final: " << t_final <<
+                          " is " << tot_diff_norm_u / tot_true_solution_u_norm << std::endl;
+                printf("Elapsed time for solving FOM: %e second\n", fom_timer.RealTime());
+                printf("Elapsed time for training DMDc: %e second\n",
+                       dmd_training_timer.RealTime());
+                printf("Elapsed time for predicting DMDc: %e second\n",
+                       dmd_prediction_timer.RealTime());
+            }
+
+            delete result_u;
+
+        }
+
+    }
+
+    // 16. Free the used memory.
+    delete ode_solver;
+    delete pmesh;
+
+    if (offline)
+    {
+        delete dmd_u;
```

Comment:
I think `dmd_u` can be deleted for the online case as well, since it is allocated in `getParametricDMDc()`.

Reply:
We should introduce smart pointers in libROM (`unique_ptr`, `shared_ptr`), which is beyond the scope of this PR. For now, let's delete pointers as needed, but I think we have too many pointer issues and need to switch to smart pointers.